### PR TITLE
Upgrade sbt to 1.2.1 and upgrade plugins

### PR DESCRIPTION
**Pull request by wsargent**
* Fix the sbt-headers references 
* Fix the sbt-scalariform changes
* Update sonatype reference with https://github.com/lightbend/config/blob/master/project/PublishToSonatype.scala
* Use `sbt-gpg` instead of `sbt-pgp`
* Left alone the `sbt-paradox` plugin because I don't understand the error messages
* change `+++` to `++` because sbt-doge is not a thing.  I don't remember what strict aggregation does, but https://github.com/sbt/sbt/issues/3698 seems to indicate it's not coming back.

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+
+addSbtPlugin("org.scalariform"       % "sbt-scalariform"     % "1.8.2")
+addSbtPlugin("com.typesafe.sbt"      % "sbt-site"            % "1.3.2")
+addSbtPlugin("de.heikoseeberger"     % "sbt-header"          % "5.0.0")
+
+addSbtPlugin("io.crashbox"           % "sbt-gpg"             % "0.2.0")
+
+// XXX Upgrading to 0.3.6 dies with an error I don't really understand right now.
 addSbtPlugin("com.lightbend.paradox" % "sbt-paradox"         % "0.2.13")
-addSbtPlugin("com.eed3si9n"          % "sbt-doge"            % "0.1.5")
```
Comment (wsargent) [inline project/plugins.sbt]: not required in sbt 1.2.1

File: project/Common.scala [hunk 2]
```diff
@@ -113,9 +117,9 @@ abstract class PublishToSonatype {
     </developers>
   }
 
-  def settings: Seq[Setting[_]] = Seq(
+  def settings = Seq(
     publishMavenStyle := true,
-    publishTo <<= isSnapshot { (snapshot) => Some(if (snapshot) ossSnapshots else ossStaging) },
+    //publishTo <<= isSnapshot { (snapshot) => Some(if (snapshot) ossSnapshots else ossStaging) },
```
Comment (wsargent) [inline project/Common.scala]: I can't figure out the error on this:

```
[error] /home/wsargent/work/ssl-config/project/Common.scala:122:50: type mismatch;
[error]  found   : Some[sbt.librarymanagement.MavenRepository]
[error]  required: sbt.Task[Option[sbt.librarymanagement.Resolver]]
[error]     publishTo <<= isSnapshot { (snapshot) => Some(if (snapshot) ossSnapshots else ossStaging) },
[error]                                                  ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
```

File: project/Common.scala [hunk 3]
```diff
@@ -49,31 +54,30 @@ object Common extends AutoPlugin {
...
+      organizationName := "Lightbend",
+      startYear := Some(2015),
+      licenses += ("Apache-2.0", new URL("https://www.apache.org/licenses/LICENSE-2.0.txt")),
 
-  def header(before: Option[String], prefix: String, after: Option[String]): String = {
-    val content = Seq("Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>")
-    (before.toSeq ++ content.map(prefix.+) ++ after.toSeq).mkString("", "\n", "\n\n")
-  }
+      HeaderPlugin.autoImport.headerLicense := Some(HeaderLicense.Custom(
+        """Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>"""
```
Comment (marcospereira) [inline project/Common.scala]: "Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>"

File: ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/Config.scala [hunk 4]
```diff
@@ -139,25 +137,24 @@ object TrustStoreConfig {
...
+final class KeyManagerConfig private[sslconfig] (
+    val algorithm:       String                        = KeyManagerFactory.getDefaultAlgorithm,
+    val keyStoreConfigs: immutable.Seq[KeyStoreConfig] = Nil) {
 
   def withAlgorithm(value: String): KeyManagerConfig = copy(algorithm = value)
   def withKeyStoreConfigs(value: scala.collection.immutable.Seq[com.typesafe.sslconfig.ssl.KeyStoreConfig]): KeyManagerConfig = copy(keyStoreConfigs = value)
 
   private def copy(
-    algorithm: String = algorithm,
+    algorithm:       String                                                                    = algorithm,
```
Comment (marcospereira) [inline ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/Config.scala]: Hum... looks like [Scalariform's `alignparameters`](https://github.com/scala-ide/scalariform#alignparameters) is behaving differently now. Maybe we should change the settings to use the default (`false`)?

Comment (wsargent) [reply ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/Config.scala]: @marcospereira should be good to go now

**Comment by marcospereira** [comment]
Thanks, @wsargent.

There is a conflict here now. I think that one last thing that could be useful here is having a task running in Travis to check both headers and code format like we have for play-json (and other projects too):

https://github.com/playframework/play-json/blob/c63c7e3d768ed9aad748b0d9acf36eb4ca01a66d/build.sbt#L267-L281

Besides that, LGTM.

**Comment by wsargent** [comment]
Added validate code

**Comment by wsargent** [comment]
@marcospereira should be good to go now

**Comment by wsargent** [comment]
@marcospereira @ktoso ping

**Pull request merged**